### PR TITLE
Add support for 'OneOf' in requestBody schemas

### LINKED ISSUE #53: OpenAPI v3: RequestBody should support OneOf, AnyOf
According to https://swagger.io/docs/specification/describing-request-body/ "OneOf" is a valid element to describe alternate schemas in the request body.

### PR DESCRIPTION
**Pull request by mhuin**
Compliant with OpenAPI v3.0

Closes #53

### CODE REVIEW

File: sphinxcontrib/openapi/__main__.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def main():
         openapi_options['examples'] = True
     if options.group:
         openapi_options['group'] = True
+    # openapi_options['request'] = True
```
Comment (ikalnytskyi) [inline sphinxcontrib/openapi/__main__.py]: What's this? A leftover?

File: sphinxcontrib/openapi/openapi30.py [hunk 2]
```diff
@@ -246,6 +248,49 @@ def _httpresource(endpoint, method, properties, convert, render_examples,
...
+                if secScheme['scheme'] == 'bearer':
+                    line = ':security: HTTP Bearer'
+                    yield '{indent}{line}'.format(**locals())
+                else:
+                    raise Exception(
+                        'Unknown http scheme "%s"' % secScheme['scheme'])
+            elif secScheme['type'] == 'apiKey':
+                key_loc = secScheme['in']
+                yield ('{indent}:security: '
+                       'API key in {key_loc}').format(**locals())
```
Comment (ikalnytskyi) [inline sphinxcontrib/openapi/openapi30.py]: I think we should also print `secScheme['name']`. Otherwise, it's not clear in which header to look in.

File: sphinxcontrib/openapi/openapi30.py [hunk 3]
```diff
@@ -246,6 +248,49 @@ def _httpresource(endpoint, method, properties, convert, render_examples,
...
 
+    securities = properties.get('security', [])
+    for security in securities:
+        yield ('{indent}This resource requires the '
+               'following authentication scheme(s):').format(**locals())
+        yield ''
+        for ref in security.keys():
+            secScheme = components.get('securitySchemes', {}).get(ref, {})
+            if secScheme['type'] == 'http':
+                if secScheme['scheme'] == 'basic':
```
Comment (ikalnytskyi) [inline sphinxcontrib/openapi/openapi30.py]: If I'm not mistaken, `secScheme['scheme']` is case insensitive. Can you please`.lower()` it before comparison? Otherwise, it won't work for specs with `Basic` spelling and that's something I'd like to address.

File: sphinxcontrib/openapi/openapi30.py [hunk 4]
```diff
@@ -246,6 +248,49 @@ def _httpresource(endpoint, method, properties, convert, render_examples,
             yield '{indent}{line}'.format(**locals())
         yield ''
 
+    securities = properties.get('security', [])
```
Comment (ikalnytskyi) [inline sphinxcontrib/openapi/openapi30.py]: It does not look like a `oneOf` support. Can you please move security schemes support to separate PR?

File: sphinxcontrib/openapi/openapi30.py [hunk 5]
```diff
@@ -264,22 +309,46 @@ def _httpresource(endpoint, method, properties, convert, render_examples,
...
-            yield '{indent}**Request body:**'.format(**locals())
-            yield ''
-            yield '{indent}.. sourcecode:: json'.format(**locals())
-            yield ''
-            for line in req_properties.splitlines():
-                # yield indent + line
-                yield '{indent}{indent}{line}'.format(**locals())
-                # yield ''
+            schs = []
+            if 'oneOf' in schema:
```
Comment (ikalnytskyi) [inline sphinxcontrib/openapi/openapi30.py]: Hm.. I wonder, is it a good idea to render every possible body input? :thinking: Can you please add tests?

Comment (mhuin) [reply sphinxcontrib/openapi/openapi30.py]: Sorry, just seeing your reviews now, I'll address them ASAP!

**Comment by mhuin** [comment]
Hello,
Anything else I need to do to get this merged?
Thanks!
